src/main/java/net/fabricmc/fabric/mixin/dimension/MixinEntity.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.dimension;
+
+import net.fabricmc.fabric.api.dimension.FabricDimensionType;
+import net.fabricmc.fabric.api.event.dimension.AttemptDimensionTeleportCallback;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.dimension.DimensionType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(Entity.class)
+public abstract class MixinEntity {
+
+    @Inject(method = "changeDimension", at = @At(value = "INVOKE_STRING", target = "Lnet/minecraft/util/profiler/Profiler;push(Ljava/lang/String;)V", args = "ldc=reposition", shift = At.Shift.AFTER), cancellable = true, locals = LocalCapture.CAPTURE_FAILHARD)
+    private void changeDimension(DimensionType targetType, CallbackInfoReturnable<Entity> cir, MinecraftServer server, DimensionType fromType, ServerWorld fromWorld, ServerWorld toWorld) {
```

Comment:
Is all this really needed! This is too much "magic" code imo.

Reply:
wdym "magic"?
thing is, vanilla already has `Entity#changeDImension()`, and uses it to transfer entities between worlds.
Because of that it wouldn#t make sense to have a static `FabricDimensionHelper.transferEntityTo()` method of sorts, because that#d mean modders would have to implement dimension changes TWICE. once for vanilla, once for fabric.

Reply:
I think this is necessary enough to keep.